lib/compress/huf_compress.c

```diff
@@ -192,9 +192,11 @@ size_t HUF_readCTable (HUF_CElt* CTable, unsigned* maxSymbolValuePtr, const void
     }   }
 
     /* fill nbBits */
+    *hasNoZeroWeights = 1;
```

Comment:
I would suggest switching this to something like:

```c
    *hasNoZeroWeights = 0;
    {   U32 n; for (n=0; n<nbSymbols; n++) {
            const U32 w = huffWeight[n];
            *hasNoZeroWeights |= w;
            CTable[n].nbBits = (BYTE)(tableLog + 1 - w) & -(w != 0);
```

It achieves the same results without any multiplications.

We should also benchmark this function before/after this change. I would suggest writing a micro-benchmark that loads some CTable repeatedly. It doesn't need to be a perfect benchmark, but we should show that the difference is small.